warp/_src/localized.py

```
@@ -0,0 +1,1587 @@
+    # Policy mode: return a function
+    if dim is None and places is None:
+
+        def cyclic_policy(dim, streams):
+            return cyclic(dim=dim, places=len(streams), block_size=block_size)
+
+        return cyclic_policy
+
+    # Direct mode: compute PartitionDesc
+    if dim is None or places is None:
+        raise ValueError("cyclic() requires either both dim and places, or neither (for policy mode)")
+    dim = tuple(int(d) for d in dim) if isinstance(dim, list) else tuple(int(d) for d in dim)
```

Ruff (0.14.2)

Useless `if`-`else` condition

(RUF034)